### PR TITLE
IBX-4929: PHP API reference

### PR DESCRIPTION
**Pull request by adriendupuis**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | [IBX-4929](https://issues.ibexa.co/browse/IBX-4929)
| Versions      | master, 4.6, ~~4.x, 3.3~~
| Edition       | Commerce (All)

- Generate and serve a PHP API reference
- Links to this PHP API reference instead of GitHub

TODO:
* [x] The whole generated reference has been updated and committed (see #1972).

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: tools/php_api_ref/phpdoc.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+
+AUTH_JSON=$1;
+OUTPUT_DIR=$2;
+
+FLAVOR='commerce';
+VERSION='4.4.*';
```
Comment (adriendupuis) [inline tools/php_api_ref/phpdoc.sh:7]:
```suggestion
VERSION='4.5.*';
```

I locally use `VERSION='4.5.*@dev';` to get the beta versions until first release.

File: tools/php_api_ref/.phpdoc/template/css/custom.css.twig [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+.phpdocumentor-title__link img {
+    width: 100%;
+    max-width: 40px;
+}
+
+.phpdocumentor .phpdocumentor-sidebar__category > * {
+    border-left-width: 0;
+}
+.phpdocumentor .phpdocumentor-sidebar {
+    overflow-x: scroll;
```
Comment (adriendupuis) [inline tools/php_api_ref/.phpdoc/template/css/custom.css.twig:46]: Have the scroll bar dispearing when not needed:
```suggestion
    overflow-x: auto;
```

File: tools/php_api_ref/phpdoc.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+set +x;
+
+AUTH_JSON=${1:-~/.composer/auth.json}; # Path to an auth.json file allowing to install the targeted edition and version
+OUTPUT_DIR=${2:-./docs/api/php_api/php_api_reference}; # Path to the directory where the built PHP API Reference is hosted
+
+DXP_EDITION='commerce'; # Edition from and for which the Reference is built
+DXP_VERSION='4.6.*@dev'; # Version from and for which the Reference is built
+DXP_EDITIONS=(oss headless experience commerce); # Available editions ordered by ascending capabilities
+TMP_DXP_DIR=/tmp/ibexa-dxp-phpdoc; # Absolute path of the temporary directory in which Ibexa DXP will be installed and the PHP API Reference built
+PHPDOC_VERSION='3.4.3'; # Version of phpDocumentor used to build the Reference
```
Comment (adriendupuis) [inline tools/php_api_ref/phpdoc.sh:12]: Found a bug in this version ([issue 3633](https://github.com/phpDocumentor/phpDocumentor/issues/3633)) that should be fixed in the next one. Wait for next [release](https://github.com/phpDocumentor/phpDocumentor/releases).

```suggestion
PHPDOC_VERSION='3.4.4'; # Version of phpDocumentor used to build the Reference
```

File: tools/php_api_ref/phpdoc.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+PHPDOC_CONF="$(pwd)/tools/php_api_ref/phpdoc.dist.xml"; # Absolute path to phpDocumentor configuration file
+PHPDOC_TEMPLATE_VERSION='3.3.1'; # Version of the phpDocumentor base template set
```
Comment (adriendupuis) [inline tools/php_api_ref/phpdoc.sh:14]: Base templates and override templates should be updated while finalizing the UI.
```suggestion
PHPDOC_TEMPLATE_VERSION='3.4.3'; # Version of the phpDocumentor base template set
```

File: tools/php_api_ref/.phpdoc/template/edition-tag.html.twig [hunk 5]
```diff
@@ -0,0 +1,18 @@
+{% macro edition_tag(node, package_edition_map) %}
+
+  {% set edition_name_map = {
+    'oss': '<abbr title="Open-source software">OSS</abbr>',
+    'content' :'Content',
```
Comment (adriendupuis) [inline tools/php_api_ref/.phpdoc/template/edition-tag.html.twig:5]:
```suggestion
    'content' :'Content',
    'headless': 'Headless',
```

File: .gitignore [hunk 6]
```diff
@@ -5,3 +5,4 @@ __pycache__/*
 **/.idea/
 .php-cs-fixer.cache
 composer.lock
+docs/api/php_api/php_api_reference/*
```
Comment (adriendupuis) [inline .gitignore:8]: :warning: Got to be removed before merge

File: tools/php_api_ref/phpdoc.sh [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+DXP_EDITIONS=(oss headless experience commerce); # Available editions ordered by ascending capabilities
+TMP_DXP_DIR=/tmp/ibexa-dxp-phpdoc; # Absolute path of the temporary directory in which Ibexa DXP will be installed and the PHP API Reference built
+PHPDOC_VERSION='3.4.3'; # Version of phpDocumentor used to build the Reference
+# PHPDOC_CONF="$(pwd)/tools/php_api_ref/phpdoc.dist.xml"; # Absolute path to phpDocumentor configuration file
+PHPDOC_CONF="$(pwd)/tools/php_api_ref/phpdoc.dev.xml"; # Absolute path to phpDocumentor configuration file
+PHPDOC_TEMPLATE_VERSION='3.4.3'; # Version of the phpDocumentor base template set
+PHPDOC_DIR="$(pwd)/tools/php_api_ref/.phpdoc"; # Absolute path to phpDocumentor resource directory (containing the override template set)
+
+PHP_BINARY="php -d error_reporting=`php -r 'echo E_ALL & ~E_DEPRECATED;'`"; # Avoid depreciation messages from phpDocumentor/Reflection/issues/529 when using PHP 8.2 or higher
+FORCE_DXP_INSTALL=0; # If 1, empty the temporary directory, install DXP from scratch, build, remove temporary directory; if O, potentially reuse the DXP already installed in temporary directory, keep temporary directory for future uses.
```
Comment (adriendupuis) [inline tools/php_api_ref/phpdoc.sh:19]: Set default mode back:

```suggestion
PHPDOC_CONF="$(pwd)/tools/php_api_ref/phpdoc.dist.xml"; # Absolute path to phpDocumentor configuration file
#PHPDOC_CONF="$(pwd)/tools/php_api_ref/phpdoc.dev.xml"; # Absolute path to phpDocumentor configuration file
PHPDOC_TEMPLATE_VERSION='3.4.3'; # Version of the phpDocumentor base template set
PHPDOC_DIR="$(pwd)/tools/php_api_ref/.phpdoc"; # Absolute path to phpDocumentor resource directory (containing the override template set)

PHP_BINARY="php -d error_reporting=`php -r 'echo E_ALL & ~E_DEPRECATED;'`"; # Avoid depreciation messages from phpDocumentor/Reflection/issues/529 when using PHP 8.2 or higher
FORCE_DXP_INSTALL=1; # If 1, empty the temporary directory, install DXP from scratch, build, remove temporary directory; if 0, potentially reuse the DXP already installed in temporary directory, keep temporary directory for future uses.
```

File: tools/php_api_ref/.phpdoc/template/index.html.twig [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{% extends 'base.html.twig' %}
+
+{% block content %}
+    <div class="main-page">
+        <h1>PHP API Reference</h1>
+
+        <p>Ibexa DXP's Public PHP API is defined in the <a href="namespaces/ibexa-contracts.html"><code>Ibexa\Contracts</code></a> namespace where consumption backward compatibility is promised.</p>
```
Comment (dabrt) [inline tools/php_api_ref/.phpdoc/template/index.html.twig:7]:
```suggestion
        <p>Ibexa DXP's public PHP API is defined in the <a href="namespaces/ibexa-contracts.html"><code>Ibexa\Contracts</code></a> namespace. Ibexa promises backward compatibility to API consumers.</p>
```

File: tools/php_api_ref/.phpdoc/template/index.html.twig [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+
+        <p>If those <code>Contracts</code> are used for type hinting (including auto-wiring), the received objects will always respect the <code>Contracts</code> in the next minor versions, following <a href="https://semver.org/">SemVer</a>.</p>
```
Comment (dabrt) [inline tools/php_api_ref/.phpdoc/template/index.html.twig:9]:
```suggestion
        <p>If those <code>Contracts</code> are used for type hinting (including auto-wiring), the received objects always respect the <code>Contracts</code> in the next minor versions, following the <a href="https://semver.org/">Semantic Versioning Specification</a>.</p>
```

**Review by dabrt — COMMENTED** [review]
Please consider applying my suggested changes.

**Pull request merged**